### PR TITLE
Plugins: show / hide install controls based on the `pluginAdminEnabled` flag

### PR DESCRIPTION
**Pull request by leventebalogh**
**Related issue:** https://github.com/grafana/grafana/issues/40503

**⚠️ &nbsp; This branch is depending on https://github.com/grafana/grafana/tree/leventebalogh/remove-deprecated-plugin-components.**

### What changed
- [x] update documentation
- [x] use the `pluginAdminEnabled` flag to show / hide install controls on the UI
- [x] add / update frontend tests

### CODE REVIEW

**Review by mckn — APPROVED** [review]
LGTM! Awesome work on this @leventebalogh

File: public/app/features/plugins/admin/components/InstallControls/InstallControls.tsx [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+  if (!hasPermission && !isExternallyManaged) {
+    const message = `You do not have permission to ${pluginStatus} this plugin.`;
+    return <div className={styles.message}>{message}</div>;
+  }
+
+  if (!plugin.isPublished) {
+    return (
+      <div className={styles.message}>
+        <Icon name="exclamation-triangle" /> This plugin is not published to{' '}
```
Comment (kaydelaney) [inline public/app/features/plugins/admin/components/InstallControls/InstallControls.tsx:71]: Unless you think the Icon here provides extra information, in which case `alt="something"` should be added to `Icon`, `alt=""` should be added so it's ignored by screen readers I think.

File: public/app/features/plugins/admin/components/InstallControls/InstallControls.tsx [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        </a>{' '}
+        and can&#39;t be managed via the catalog.
+      </div>
+    );
+  }
+
+  if (!isCompatible) {
+    return (
+      <div className={styles.message}>
+        <Icon name="exclamation-triangle" />
```
Comment (kaydelaney) [inline public/app/features/plugins/admin/components/InstallControls/InstallControls.tsx:83]: Same as above

**Review by kaydelaney — APPROVED** [review]
Looks good, just a small accessibility nit

File: public/app/features/plugins/admin/components/InstallControls/InstallControls.tsx (hunk 1 above)
Comment (leventebalogh) [reply public/app/features/plugins/admin/components/InstallControls/InstallControls.tsx:71]: Thanks @kaydelaney, that's a good point 👍

Comment (mckn) [reply public/app/features/plugins/admin/components/InstallControls/InstallControls.tsx:71]: Nice, I didn't know this. Thanks for pointing this out @kaydelaney  🙏🏻

Comment (leventebalogh) [reply public/app/features/plugins/admin/components/InstallControls/InstallControls.tsx:71]: @kaydelaney I hope you don't mind, but I probably will follow up on these in a separate issue as I am a bit uncertain about how to do it properly. (messaged you on Slack as well, will update this thread with the outcome of our conversation).

Issue for this: https://github.com/grafana/grafana/issues/41947

**Pull request merged**